video.cpp

```diff
@@ -382,29 +406,50 @@ static bool read_video_filter(int type, VideoFilter *out)
 			is_adaptive ? count / 2 : count,
 			is_adaptive ? "true" : "false" );
 
+	bool valid = false;
 	if (is_adaptive)
 	{
 		out->is_adaptive = true;
-		bool valid = scale_phases(out->phases, phases, count / 2);
+		valid = scale_phases(out->phases, phases, count / 2);
 		valid = valid && scale_phases(out->adaptive_phases, phases + (count / 2), count / 2);
-		return valid;
 	}
 	else if (count == 32 && !is_adaptive) // legacy
 	{
 		out->is_adaptive = false;
-		return scale_phases(out->phases, phases, 16);
+		valid = scale_phases(out->phases, phases, 16);
 	}
 	else if (!is_adaptive)
 	{
 		out->is_adaptive = false;
-		return scale_phases(out->phases, phases, count);
+		valid = scale_phases(out->phases, phases, count);
 	}
+	else
+	{
+		// Make a default NN filter in case of error
+		out->is_adaptive = false;
+		FilterPhase nn_phases[2] =
+		{
+			{ .t = { 0, 256, 0, 0 } },
+			{ .t = { 0, 0, 256, 0 } }
+		};
+		scale_phases(out->phases, nn_phases, 2);
+		valid = false;
+	}
```

Comment:
this part of the code is unreachable.
Is that supposed to kick in when is_adaptive and valid is false?
I would suggest the following refactoring:
```c
bool valid;
if(is_adaptive) {
	out->is_adaptive = true;
	valid = scale_phases(out->phases, phases, count / 2);
	valid = valid && scale_phases(out->adaptive_phases, phases + (count / 2), count / 2);
} else {
	out->is_adaptive = false;
	valid = scale_phases(out->phases, phases, count == 32 ? 16 : count);
	if(!valid){
		// Make a default NN filter in case of error
		FilterPhase nn_phases[2] = {
			{.t = {0, 256, 0, 0}},
			{.t = {0, 0, 256, 0}}
		};
		scale_phases(out->phases, nn_phases, 2);
	}
}
```

Reply:
Thanks, I'll fix it up. It is meant to be a fallback for when parsing fails for any reason.